solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -301,6 +301,8 @@ && getZkController().getOverseer() != null
 
   private final AllowListUrlChecker allowListUrlChecker;
 
+  private final boolean hideStackTrace;
```

Comment:
I don't think a simple global boolean needs to be yet another field on CoreContainer.  Even the NodeConfig stuff is overkill for what could be a system property.

If we do keep NodeConfig, we could still do away with this field because the NodeConfig stays around as a field on CoreContainer as `cfg`.